src/lib/php/Report/ReportUtils.php

```diff
@@ -287,4 +287,34 @@ public function getFilesWithLicensesFromClearings(
     }
     return $filesWithLicenses;
   }
+
+  /**
+   * Upsert an entry in the reportgen table.
+   *
+   * This function checks if an entry with the specified upload_fk and filepath
+   * exists in the reportgen table. If it exists, it updates the job_fk for that row.
+   * If it does not exist, a new row is inserted with the provided values.
+   *
+   * @param int $upload_fk  Foreign key for the upload.
+   * @param int $job_fk     ID of the job to be updated or inserted.
+   * @param string $filepath  Filepath associated with the entry.
+   */
+  public function upsertReportgenEntry($upload_fk, $job_fk, $filepath)
```

Comment:
Please use more common terms for naming the functions. `upsert` is not used vert commonly and will make it difficult to maintain the code.

Reply:
Sure.